lib/src/component_declaration/component_base.dart

```diff
@@ -67,16 +69,59 @@ ReactDartComponentFactoryProxy registerComponent(react.Component dartComponentFa
   return reactComponentFactory;
 }
 
+/// Helper function that wraps react.registerComponent2, and allows attachment of additional
+/// component factory metadata.
+///
+/// * [isWrapper]: whether the component clones or passes through its children and needs to be
+/// treated as if it were the wrapped component.
+///
+/// * [builderFactory]/[componentClass]: the [UiFactory] and [UiComponent2] members to be potentially
+/// used as types for [isComponentOfType]/[getComponentFactory].
+///
+/// * [displayName]: the name of the component for use when debugging.
+ReactDartComponentFactoryProxy2 registerComponent2(react.Component2 dartComponentFactory(), {
+    bool isWrapper: false,
+    ReactDartComponentFactoryProxy2 parentType,
+    UiFactory builderFactory,
+    Type componentClass,
+    String displayName,
+    Iterable<String> skipMethods = const ['getDerivedStateFromError', 'componentDidCatch'],
+}) {
+  ReactDartComponentFactoryProxy2 reactComponentFactory = react.registerComponent(dartComponentFactory, skipMethods);
+
+  if (displayName != null) {
+    reactComponentFactory.reactClass.displayName = displayName;
+  }
+
+  registerComponentTypeAlias(reactComponentFactory, builderFactory);
+  registerComponentTypeAlias(reactComponentFactory, componentClass);
+
+  setComponentTypeMeta(reactComponentFactory, isWrapper: isWrapper, parentType: parentType);
+
+  return reactComponentFactory;
+}
+
 /// Helper function that wraps [registerComponent], and allows an easier way to register abstract components with the
 /// main purpose of type-checking against the abstract component.
 ///
 /// __The result must be stored in a variable that is named very specifically:__
 ///
 ///     var $`AbstractComponentClassName`Factory = registerAbstractComponent(`AbstractComponentClassName`);
-// ignore: deprecated_member_use
+///
+/// __Deprecated.__ Use [registerAbstractComponent2] instead.
+@Deprecated('4.0.0')
 ReactDartComponentFactoryProxy registerAbstractComponent(Type abstractComponentClass, {ReactDartComponentFactoryProxy parentType}) =>
     registerComponent(() => new DummyComponent(), componentClass: abstractComponentClass, parentType: parentType);
 
+/// Helper function that wraps [registerComponent2], and allows an easier way to register abstract components with the
+/// main purpose of type-checking against the abstract component.
+///
+/// __The result must be stored in a variable that is named very specifically:__
+///
+///     var $`AbstractComponentClassName`Factory = registerAbstractComponent2(`AbstractComponentClassName`);
+ReactDartComponentFactoryProxy2 registerAbstractComponent2(Type abstractComponentClass, {ReactDartComponentFactoryProxy2 parentType}) =>
```

Comment:
Sorry, we can probably take this and the `registerAbstractComponent` deprecation - its outside the scope of this PR.  That's my bad.